### ISSUE TITLE
commas_linter() could allow spaces before commas to match air

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3053]
Possibly related/subsumes: #2442 

`air` prefers spaces _before_ a comma when doing vertical alignment, where {styler} prefers spaces _after_ a comma, e.g.

```
# Formatted by air
tibble::tribble(
  ~x       , ~y ,
  "another", 1:3 ,
  "b"      , 1211234
)

# Formatted by styler
tibble::tribble(
  ~x,        ~y,
  "another", 1:3,
  "b",       1211234
)
```

I suspect it's possible to just support this by default without an option by checking that, for a given possible lint on 'spaces before', all lines in that call with at least the same line length _also_ have a comma in the same position. It can let through a few false negatives but I think it will be rare.

Alternatively, we could just make this an option like `allow_before` that totally skips checking spacing before commas. This approach will let in a lot more "false negatives" in the sense that a formatter will make a change that the linter did not find.
